src/middleware.ts

```diff
@@ -1,16 +1,43 @@
 import { clerkMiddleware } from "@clerk/nextjs/server";
-
-export default clerkMiddleware();
+import { NextRequest, NextResponse, NextFetchEvent } from "next/server";
 
 /**
  * Clerk middleware causes page latency, we're only enabling it for API routes or pages where
  * we need to access the user session in the backend
+ *
+ * The middleware also handles adding X-Robots-Tag: noindex for URLs with sid parameters
+ * to prevent crawl/index bloat from session IDs
  */
+export default function middleware(req: NextRequest, ev: NextFetchEvent) {
+  const url = new URL(req.url);
+  const hasSidParam = Array.from(url.searchParams.keys()).some((key) =>
+    key.startsWith("sid"),
+  );
+
+  // Only run Clerk on API/TRPC routes
+  if (url.pathname.startsWith("/api") || url.pathname.startsWith("/trpc")) {
+    const handler = clerkMiddleware();
+    const result = handler(req, ev);
+
+    // Ensure header is present even when Clerk handles the response
+    return Promise.resolve(result).then((res) => {
+      if (hasSidParam && res) {
+        res.headers.set("X-Robots-Tag", "noindex");
+      }
+      return res;
+    });
+  }
+
+  const res = NextResponse.next();
+  if (hasSidParam) {
+    res.headers.set("X-Robots-Tag", "noindex");
+  }
+  return res;
+}
+
 export const config = {
   matcher: [
-    // Skip Next.js internals and all static files, unless found in search params
-    // "/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)",
-    // Always run for API routes
-    "/(api|trpc)(.*)",
+    // Run on all routes except Next.js internals and static assets
+    "/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)",
```

Comment:
Doesn't this mean that previously statically generated routes are now dynamic routes? 

So we're essentially changing all the OWA routes into dynamic routes?